src/gtk/window.cpp

```diff
@@ -54,6 +55,10 @@ using namespace wxGTKImpl;
 typedef guint KeySym;
 #endif
 
+#if defined (GDK_WINDOWING_X11) || defined (GDK_WINDOWING_WAYLAND)
```

Comment:
You need to test for `HAVE_XKBCOMMON` too here to avoid compilation problems with GTK 2 where libxkbcommon might not be available.

Moreover, to avoid repeating the same complicated preprocessor test condition again below, I suggest adding
```cpp
#define wxHAS_XKB
```
here and then just doing `#ifdef wxHAS_XKB` later.